cassiopeia/data/CassiopeiaTree.py

```diff
@@ -1503,38 +1503,43 @@ def add_leaf(
         # This function will also clear the cache
         self.__register_data_with_tree()
 
-    def remove_leaf_and_prune_lineage(self, node: str) -> None:
-        """Removes a leaf from the tree and prunes the lineage.
+    def remove_leaves_and_prune_lineages(
+        self, nodes: Union[str, List[str]]
+    ) -> None:
+        """Removes a leaf (leaves) from the tree and prunes the lineage(s).
 
-        Removes a leaf and all ancestors of that leaf that are no longer the
-        ancestor of any leaves. In the context of a phylogeny, this prunes the
-        lineage of all nodes no longer relevant to observed samples.
-        Additionally, maintains consistency with the updated tree by removing
-        the node from all leaf data.
+        Removes the specified leaves and all ancestors of those leaves that are
+        no longer the ancestor of any of the remaining leaves. In the context
+        of a phylogeny, this prunes the lineage of all nodes no longer relevant
+        to observed samples. Additionally, maintains consistency with the
+        updated tree by removing the node from all leaf data.
 
         Args:
-            node: The leaf node to be removed
+            nodes: The leaf (leaves) to be removed
```

Comment:
Please be clear that this can take in either a single leaf, as a string, or a list of leaves.